apps/launch/hooks/useGetMyStakingContracts.ts

```diff
@@ -0,0 +1,175 @@
+  const { chainId } = useAccount();
+  const [metadata, setMetadata] = useState<Metadata[]>([]);
+
+  const contracts = addresses.map((address) => ({
+    address,
+    abi: STAKING_TOKEN.abi as Abi,
+    chainId,
+    functionName: 'metadataHash',
+    args: [],
+  }));
+
+  const { data } = useReadContracts({
+    contracts,
+    query: {
+      enabled: addresses.length > 0,
+      select: async (list) => {
+        const metadataHashList = list.map(({ result }) => result as string);
+        try {
+          const metadataList = await Promise.allSettled<Metadata>(
+            metadataHashList.map((hash) => getMetadata(hash)),
+          );
+
+          const results: Metadata[] = [];
+          for (const response of metadataList) {
+            if (response.status === 'fulfilled' && response.value) {
+              results.push(response.value);
+            }
+          }
+
+          return results.filter((metadata) => !!metadata);
+        } catch (error) {
+          window.console.error(error);
+        }
+
+        return [];
+      },
+    },
+  });
+
+  useEffect(() => {
+    (async () => {
+      if (!data) return;
+
+      try {
+        const result = await data;
+        setMetadata(result);
+      } catch (error) {
+        window.console.error(error);
+      }
+    })();
+  }, [data, addresses]);
+
+  return metadata;
+};
+
+const useGetInstanceAddresses = () => {
+  const client = usePublicClient();
+  const { chainId } = useAccount();
+
+  const [instanceAddresses, setInstanceAddresses] = useState<Address[]>([]);
+
+  const currentChainId = chainId as SupportedChain;
+
+  useEffect(() => {
+    (async () => {
+      if (!currentChainId) return;
+      if (!client) return;
+
+      try {
+        const eventLogs = await client.getLogs({
+          address: STAKING_FACTORY.addresses[`${currentChainId}`] as Address,
+          event: parseAbiItem(
+            'event InstanceCreated(address indexed, address indexed, address indexed)',
+          ),
+          fromBlock: BigInt(0),
+          toBlock: 'latest',
+        });
+
+        // log.args[1] is the instance address
+        const addresses = eventLogs.map((log) => log.args[1] as Address);
+        setInstanceAddresses(addresses);
+      } catch (e) {
+        window.console.error(e);
+      }
+    })();
+  }, [currentChainId, client]);
+
+  return instanceAddresses;
+};
+
+export const useGetMyStakingContracts = () => {
+  const dispatch = useAppDispatch();
+  const instanceAddresses = useGetInstanceAddresses();
+  const myStakingContractsMetadata = useGetMyStakingContractsMetadata(instanceAddresses);
+  const nominees = useGetAllNominees();
+
+  const instanceAddressesInBytes32 = instanceAddresses.map((address) =>
+    getBytes32FromAddress(address),
+  );
+
+  useEffect(() => {
+    if (!myStakingContractsMetadata) return;
+    if (myStakingContractsMetadata.length === 0) return;
+    if (!nominees) return;
```

Comment:
can we also check that the myStakingContracts is empty, this hook will be triggered again on each page change and without this check the data will be recalculated and dispatched again